extensions/amazon-aws/src/ecs.tsx

```diff
@@ -1,62 +1,35 @@
-import { Cluster, DescribeClustersCommand, ECSClient, ListClustersCommand } from "@aws-sdk/client-ecs";
-import { ActionPanel, List, Action, Icon } from "@raycast/api";
+import { Icon, List } from "@raycast/api";
 import { useCachedPromise } from "@raycast/utils";
-import AWSProfileDropdown, { AWS_URL_BASE } from "./aws-profile-dropdown";
+import { fetchClusters } from "./actions";
+import AWSProfileDropdown from "./components/searchbar/aws-profile-dropdown";
+import { getFilterPlaceholder } from "./util";
+import ECSCluster from "./components/ecs/ECSCluster";
+import { useState } from "react";
 
 export default function ECS() {
-  const { data: clusters, error, isLoading, revalidate } = useCachedPromise(fetchClusters);
+  const [isServicesLoading, setServicesLoading] = useState<boolean>(false);
+  const {
+    data: clusters,
+    error,
+    isLoading,
+    revalidate,
+  } = useCachedPromise(fetchClusters, [], { keepPreviousData: true });
+
+  if (error) {
+    return <List.EmptyView title={error.name} description={error.message} icon={Icon.Warning} />;
+  }
 
   return (
     <List
-      isLoading={isLoading}
-      searchBarPlaceholder="Filter clusters by name..."
+      isLoading={isLoading || isServicesLoading}
+      searchBarPlaceholder={getFilterPlaceholder("clusters")}
       searchBarAccessory={<AWSProfileDropdown onProfileSelected={revalidate} />}
     >
-      {error ? (
-        <List.EmptyView title={error.name} description={error.message} icon={Icon.Warning} />
+      {clusters ? (
```

Comment:
This will not work when the returned value is an empty array (which happens in my case).
```suggestion
      {clusters && clusters.length > 0 ? (
```